pkg/collector/osmlogs_collector.go

```diff
@@ -123,6 +133,61 @@ func collectPodConfigs(collector *OsmLogsCollector, rootPath, meshName, namespac
 	return nil
 }
 
+// ** Collect Envoy proxy config for pods in monitored namespace: port-forward and curl config dump **
+func collectProxyConfigs(collector *OsmLogsCollector, rootPath, meshName, namespace string) error {
+	pods, err := getResourceList([]string{"get", "pods", "-n", namespace, "-o", "jsonpath={..metadata.name}"}, " ")
+	if err != nil {
+		return err
+	}
+	for _, podName := range pods {
+		pid, err := utils.RunBackgroundCommand("kubectl", "port-forward", podName, "-n", namespace, "15000:15000")
+		if err != nil {
+			log.Printf("Failed to collect Envoy config for pod %s in OSM monitored namespace %s: %+v", podName, namespace, err)
+			continue
+		}
+
+		var (
+			resp    *http.Response
+			retries int = 5
+		)
+		for retries > 0 {
+			resp, err = http.Get("http://localhost:15000/config_dump")
+			if err != nil {
+				retries -= 1
+				log.Printf("Error curling config dump: %+v. %d retries remaining...\n", err, retries)
+				time.Sleep(5 * time.Second)
```

Comment:
Why do we need to sleep here?

Reply:
was facing an error where the curl was occurring before the port forwarding started, which is why there's this whole sleep and retry loop